staging_vespalib/src/vespa/vespalib/metrics/simple_metrics_manager.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <memory>
+#include <thread>
+#include <vespa/vespalib/stllike/string.h>
+#include "name_collection.h"
+#include "mergers.h"
+#include "snapshots.h"
+#include "metrics_manager.h"
+#include "metric_types.h"
+#include "clock.h"
+
+namespace vespalib {
+namespace metrics {
+
+struct SimpleManagerConfig {
+    int sliding_window_seconds;
+    // possibly more config later
+    SimpleManagerConfig() : sliding_window_seconds(60) {}
+};
+
+
+/**
+ * Simple manager class that puts everything into a
+ * single global repo with std::mutex locks used around
+ * most operations.  Only implements sliding window
+ * and a fixed (1 Hz) collecting interval.
+ * Consider renaming to "LockingManager" or "SlidingWindowManager".
+ **/
+class SimpleMetricsManager : public MetricsManager
```

Comment:
is it possible to decompose 'metrics collection', 'bucket management' and 'time handling' more? (it is harder to test with an internal thread doing 1s ticks...)